Assurance Cases.md

```diff
@@ -13,12 +13,24 @@
 
 ![Assurance Case 1 Diagram:](/images/Assurance_Case_1.PNG)
 
 #### Evidence for Assurance Case 1
-.red[TODO] Insert Evidence 1
+Home Assistant offers users the ability to edit the `configuration.yaml` file on the local network via a variety of different options. The `configuration.yaml` file is the base configuration file where users can add their [YAML]( https://yaml.org/) code to specify device settings. This gives users the ability to integrate devices that are not configurable through Home Assistants user interface. The `configuration.yaml` files is a plain-text file that is readable by anyone who has access to the file, as seen in rebuttal R1. The file contains API tokens, account usernames, and passwords. As seen in sub-claim C2, Home Assistant recommends users store their sensitive information in the [secrets.yaml]( https://www.home-assistant.io/docs/configuration/secrets/) file, which is a separate configuration file. `Secrets.yaml` allows users to replace sensitive information in the `configuration.yaml` file with `!secret`. `!secret` will tell the configuration file that the sensitive information is in `secrets.yaml`. However as seen in rebuttal R2, `secrets.yaml` still stores password information in plain test and resides in the same `config` folder as the `configuration.yaml` file. If the `config` folder is comprised, a malicious user will have access to sensitive account information and API keys. 
+
+Sub-claim C3 and Evidence E1 and E2 help protect sensitive information by giving users different locations to store sensitive account information that is not in plain text. While `secrets.yaml` is mentioned in the Home Assistant setup documentation, they also mention [Keyrings]( https://www.home-assistant.io/docs/tools/keyring/) and [Credstash]( https://www.home-assistant.io/docs/tools/credstash/). E2 protects sensitive user data because there is no longer a `secrets.yaml` file in the base `config` folder. As seen in E1, sensitive data is replaced with `!secret` and will grab the sensitive data from Credstash or the Keyring. Credstash and Keyrings can still be compromised as seen in Undermine UM1. Sub-claims C4 focuses on Keyring and explains how sensitive information is stored on the Operating System and only accessible through password authentication (Evidence E3). Sub-claim C5 focuses on using Credstash which requires an Amazon Webservices (AWS) account set up and accounts must follow AWS password [guidelines]( https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_passwords_change-root.html) (Evidence E4). The passwords in E3 and E4 will protect sensitive user data from being compromised by a malicious user. Both Evidence E3 and E4 and their respective sub-claims allow more protection than what was offered in sub-claim C2. Now the standard user will not have to store a password in plaintext files that can be accessible by anyone on the local network.
```

Comment:
All looks good. The only thing I have a question on is about the keyring. One limitation of this is that it asks for the password at start up so it could bring things down when power is lost or other abrupt issues happen. This might make people not implement features like the keyring. Here is a link of a user complaining about this and leaving his passwords in plain text to avoid outages with HA. Not sure if this is something HA can address. 
https://community.home-assistant.io/t/how-to-use-keyring-in-home-assistant-hassos/212092/8

Reply:
Yeah, I think that will be more of an inconvenience over a security flaw. If the device is powered down the keyring is inaccessible because it's stored on the OS.  I don't believe that will be a limitation that will cause a user to compromise sensitive account information because once the device boots back up it will still prompt for the password.